### PR TITLE
2638 upgrade spring boot

### PR DESCRIPTION
**Pull request by ConorH994**
https://tools.hmcts.net/jira/browse/AM-2638

### CODE REVIEW

**Review by sankar-padakula — APPROVED** [review]

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -30,12 +30,19 @@ security:
     - "/loggers/**"
     - "/am/role-assignments/fetchFlagStatus"
     - "/swagger-ui.html"
+    - "/swagger-ui/**"
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/resources/application.yaml:33]: Resolved in #1447 see d23d942342833ba0c47942465bff577277331ec6

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -30,12 +30,19 @@ security:
...
+    - "/v3/api-docs"
+    - "/v3/**"
     - "/webjars/springfox-swagger-ui/**"
     - "/search/**"
     - "/csrf"
     - "/error"
+    - "/status/health"
+    - "/welcome"
+    - "/health/liveness"
+    - "/loggers/**"
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/resources/application.yaml:44]: Resolved in #1447 see d23d942342833ba0c47942465bff577277331ec6

File: src/main/java/uk/gov/hmcts/reform/orgrolemapping/config/SwaggerConfiguration.java [hunk 3]
```diff
@@ -1,68 +1,51 @@
...
+                    .in(ParameterIn.HEADER.toString())
+                    .schema(new StringSchema())
+                    .name("ServiceAuthorization")
+                    .description("Keyword `Bearer` followed by a service-to-service token "
+                            + "for a whitelisted micro-service")
+                    .required(true);
+            Parameter authorizationHeader = new Parameter()
+                    .in(ParameterIn.HEADER.toString())
+                    .schema(new StringSchema())
+                    .name("Authorization")
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/reform/orgrolemapping/config/SwaggerConfiguration.java:43]: Resolved in #1447 see d23d942342833ba0c47942465bff577277331ec6

File: src/main/java/uk/gov/hmcts/reform/orgrolemapping/config/SwaggerConfiguration.java [hunk 4]
```diff
@@ -1,68 +1,51 @@
...
+                    .description("Authorization token")
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/reform/orgrolemapping/config/SwaggerConfiguration.java:44]: Resolved in #1447 see d23d942342833ba0c47942465bff577277331ec6

File: src/main/java/uk/gov/hmcts/reform/orgrolemapping/controller/FeatureFlagController.java [hunk 5]
```diff
@@ -33,6 +37,12 @@ public ResponseEntity<Object> getFeatureFlag(@RequestParam(value = "flagName") S
             produces = MediaType.APPLICATION_JSON_VALUE,
             consumes = {"application/json"}
     )
+    @Operation(summary = "Create feature flag",
+            security =
+                    {
+                            @SecurityRequirement(name = "Authorization"),
+                            @SecurityRequirement(name = "ServiceAuthorization")
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/reform/orgrolemapping/controller/FeatureFlagController.java:44]: Resolved in #1447 see d23d942342833ba0c47942465bff577277331ec6

File: src/main/java/uk/gov/hmcts/reform/orgrolemapping/controller/RefreshController.java [hunk 6]
```diff
@@ -47,21 +50,25 @@ public RefreshController(RefreshOrchestrator refreshOrchestrator,
             produces = V1.MediaType.MAP_ASSIGNMENTS,
             consumes = {"application/json"}
     )
+    @Operation(summary = "refresh",
+            security =
+                    {
+                            @SecurityRequirement(name = "Authorization"),
+                            @SecurityRequirement(name = "ServiceAuthorization")
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/reform/orgrolemapping/controller/RefreshController.java:57]: Resolved in #1447 see d23d942342833ba0c47942465bff577277331ec6

File: src/main/java/uk/gov/hmcts/reform/orgrolemapping/controller/RefreshController.java [hunk 7]
```diff
@@ -76,22 +83,25 @@ public ResponseEntity<Object> refresh(@RequestParam Long jobId,
...
-            @ApiResponse(
-                    code = 422,
-                    message = V1.Error.UNPROCESSABLE_ENTITY_REQUEST_REJECTED
-            )
-    })
+    @Operation(summary = "refreshes judicial role assignments",
+            security =
+                    {
+                            @SecurityRequirement(name = "Authorization"),
+                            @SecurityRequirement(name = "ServiceAuthorization")
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/reform/orgrolemapping/controller/RefreshController.java:90]: Resolved in #1447 see d23d942342833ba0c47942465bff577277331ec6

File: src/main/java/uk/gov/hmcts/reform/orgrolemapping/controller/WelcomeController.java [hunk 8]
```diff
@@ -49,18 +54,21 @@ public String welcome() {
...
-            @ApiResponse(
-                    code = 400,
-                    message = V1.Error.INVALID_REQUEST
-            )
-    })
+    @Operation(summary = "creates multiple role assignments based upon user profile mapping rules",
+            security =
+                    {
+                            @SecurityRequirement(name = "Authorization"),
+                            @SecurityRequirement(name = "ServiceAuthorization")
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/reform/orgrolemapping/controller/WelcomeController.java:61]: Resolved in #1447 see d23d942342833ba0c47942465bff577277331ec6

File: src/main/java/uk/gov/hmcts/reform/orgrolemapping/util/Swagger2SpringBoot.java [hunk 9]
```diff
@@ -1,14 +1,18 @@
...
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.context.annotation.ComponentScan;
-import springfox.documentation.swagger2.annotations.EnableSwagger2WebMvc;
 
 @SpringBootApplication
-@EnableSwagger2WebMvc
+@SecurityScheme(name = "Authorization", type = SecuritySchemeType.HTTP, bearerFormat = "JWT", scheme = "bearer")
+@SecurityScheme(name = "ServiceAuthorization", type = SecuritySchemeType.APIKEY,
+        in = SecuritySchemeIn.HEADER, bearerFormat = "JWT", description = "ServiceAuthorization")
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/reform/orgrolemapping/util/Swagger2SpringBoot.java:15]: Resolved in #1447 see d23d942342833ba0c47942465bff577277331ec6

NB: class couldn't be removed without breaking integration tests. So one to review in future ticket.

**Review by mattnayler — APPROVED** [review]

**Review by mattnayler — APPROVED** [review]
Re-reviewed at 02699f40ee0fa8b48d5ca68d966fcb9991efb205

File: src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/provider/OrgRoleMappingProviderTestForRefresh.java [hunk 10]
```diff
@@ -80,6 +80,7 @@ void pactVerificationTestTemplate(PactVerificationContext context) {
     @BeforeEach
     void beforeCreate(PactVerificationContext context) {
         MockMvcTestTarget testTarget = new MockMvcTestTarget();
+        //System.getProperties().setProperty("pact.verifier.publishResults", "true");
```
Comment (tomxelliott) [inline src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/provider/OrgRoleMappingProviderTestForRefresh.java:83]: Can this be removed? If not shall we add a comment so others know why it is commented out?

**Review by tomxelliott — APPROVED** [review]

File: src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/provider/OrgRoleMappingProviderTestForRefresh.java (hunk 10 above)
Comment (mikebrownccd) [reply src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/provider/OrgRoleMappingProviderTestForRefresh.java:83]: further comment added with reason for inclusion. resolved

**Pull request merged**